### DISCUSSION TITLE
Is there a python interperter that can work with the cc65 compiler?

### DISCUSSION THREAD

**Discussion by Nightchanger** [cc65/cc65#2421] [General]
I'm making an 8bit system using the 6502 cpu instruction set. And to simplify the system to those around me who are familliar with python I'm looking for an interperter I can use to compile it to the cpu. So far the cc65 looks very promising, and I've been trying to adjust the cython superset to do so. But being tedious just on paper gives me concern. Is there a more simplified way I haven't thought about?

**Comment by rofl0r** [comment]
the closest to python you can get is probably https://github.com/keith-packard/snek .
even micropython seems overkill for 6502. i don't know what your intention with cython was, but cython coexists with the regular cpython install which requires megabytes of space on disk, and probably also in ram.

**Comment by Nightchanger** [comment]
I'm doing a project in making an 8bit system based on the 6502 cpu. I'm
more familliar with python and most of my college as well, that I prefer to
use it as I'll get more help.
I'm right now writing my project plan and part of it in the ends is to make
an sdk that includes a config file to write python scripts to it. But seen
how heavy cython is I'm using it to prototype a chip-8 interpreter and the
emulator for the system.

Thanks for pointing out snek for me. It looks very promising. Hopefully I
can integrate it.

‫בתאריך יום ד׳, 28 בפבר׳ 2024 ב-15:00 מאת ‪rofl0r‬‏ <‪
***@***.***‬‏>:‬
